### ISSUE TITLE
[Bug Report] Insufficient initialization in `copy_tile_to_dst_init_short_with_dt`

### ISSUE THREAD

**Issue by wenbinlyuTT** [tenstorrent/tt-metal#19453]
**Describe the bug**
When adding new quantization ops built on top of BinaryNg, I discovered that under some specific input data format combinations, the CB-to-DEST copy performed by the unpacker will result in incorrect results in DEST. So far I've been able to trigger this when the data formats in DEST0 and DEST1 have different sizes (e.g. i32 input in DEST0, bf16 input in DEST1, and expected i32 output in DEST0).

Replacing the `copy_tile_to_dst_init_short_with_dt` immediately before the `copy_tile` invocation with `unary_op_init_common` solves the problem. Indicating that there might be something missing from the more lightweight `copy_tile_to_dst_init_short_with_dt` configuration call.

**To Reproduce**
1. Build the reproducer branch at [wenbinlyu/a2d_init_bug](https://github.com/tenstorrent/tt-metal/tree/wenbinlyu/a2d_init_bug) (if branch is gone, commit https://github.com/tenstorrent/tt-metal/commit/f61154185292ff428bf3922ae1405a47b100525b)
2. Run the reproducer below and observe that the re-quant result is some constant (so effectively scale == 0)
3. Navigate to `ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/kernels/compute/eltwise_binary_sfpu_scalar.cpp`, follow the comments to replace the init calls above the two `copy_tile` calls, run the reproducer again and see the issue disappear.
4. Additionally, run`pytest tests/ttnn/unit_tests/operations/eltwise/test_quantization.py` to verify the accuracy of the quant ops, especially for larger input tensors that requires better numerical stability.

**Reproducer**
```
import torch
import ttnn

def calculate_scale_zero_point(torch_input_tensor, q_min, q_max):
    i_min = torch.min(torch_input_tensor).item()
    i_max = torch.max(torch_input_tensor).item()
    scale = (i_max - i_min) / (q_max - q_min)
    zero_point = q_min - int(i_min / scale)
    return (scale, zero_point)

dev = ttnn.open_device(device_id=0)

N = 3
tensor_tr = torch.rand(N, N, dtype=torch.float32)
tensor_tt = ttnn.from_torch(tensor_tr, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=dev)
print("++ Input:\n", tensor_tt)

scale, zero_point = calculate_scale_zero_point(tensor_tr, -128, 127)
print("++ scale", scale)
print("++ 1/scale", 1 / scale)
print("++ zero_point", zero_point)

tensor_quantized_tt = ttnn.quantize(tensor_tt, scale, zero_point)
print("++ Quantized:\n", tensor_quantized_tt)

scale_r, zero_point_r = calculate_scale_zero_point(tensor_tr, -37, 73)
print("++ scale_r", scale_r)
print("++ 1/scale_r", 1 / scale_r)
print("++ zero_point_r", zero_point_r)

tensor_requantized_tt = ttnn.requantize(tensor_quantized_tt, scale, zero_point, scale_r, zero_point_r)
print("++ Re-Quantized:\n", tensor_requantized_tt)

tensor_derequantized_tt = ttnn.dequantize(tensor_requantized_tt, scale_r, zero_point_r)
print("++ De-Re-Quantized:\n", tensor_derequantized_tt)

ttnn.close_device(dev)
```

**Comment by rdjogoTT** [comment]
The Dest register will not be switching modes, it is only ever in 32bit mode. Bfloat16 can safely be unpacked and put into Dest while it is in 32bit mode, because it will be stored as a float32 value but with appropriate bits set to 0 (this already works for other OPs).

The problem here is that the `copy_tile_to_dst_init_short_with_dt` calls on the `reconfig_data_format` API which does not yet support reconfiguring to/from int32 formats. The `unary_op_init_common` call does a complete config of the HW which is why it solves the problem. The solution is to add only the missing pieces from `unary_op_init_common` to the `reconfig_data_format` API to enable int32.

**Comment by minaliuTT** [comment]
@vmilicevicTT Could you confirm if this is still a P2 bug, or can it be closed?

**Comment by nvelickovicTT** [comment]
@ncvetkovicTT can you tell if this is already covered with the refactor? Is reconfig supporting int32 format?

**Comment by ncvetkovicTT** [comment]
Although a thorough refactoring of reconfig calls will happen incrementally, I believe that this particular issue will be resolved with this ticket (draft PRs are in place) https://github.com/tenstorrent/tt-metal/issues/35079.

@njokovicTT You and I can run the repro to check if the issue persists (if the branch still exists :/ ) once you merge the ALU config fix.

**Comment by vmilicevicTT** [comment]
@ncvetkovicTT , @njokovicTT , can you confirm if we can close this?  Thanks

**Comment by ncvetkovicTT** [comment]
Checked this on a Wormhole n150 with the latest code. Built it, ran the original repro, and the re-quant output now comes back correct (not a constant anymore). Also ran the full quantization test suite (test_quantization.py + test_quantization2.py) and all 8808 cases pass.

The int32 reconfig in `copy_tile_to_dst_init_short_with_dt` works now, which was the actual root cause here. This got fixed as part of #35079. Closing.
